File: 443-string-compression.ts
function compress(chars: string[]): number {
  let index = 0
  let writeIndex = 0
  let count = 0

  while (index < chars.length) {
    count++
    const char = chars[index]
    const nextChar = chars[index + 1]
    const isLast = char !== nextChar
    if (isLast) {
      chars[writeIndex++] = char
      if (count >= 2) {
        const s = `${count}`
        for (const c of s) {
          chars[writeIndex++] = c
        }
      }
      count = 0
    }
    index++
  }

  return writeIndex
}

File: 443-string-compression.py
class Solution:
    def compress(self, chars: list[str]) -> int:
        index = 0
        write_index = 0
        count = 0

        while index < len(chars):
            count += 1
            char = chars[index]
            next_char = chars[index + 1] if index < len(chars) - 1 else None
            is_last = char != next_char
            if is_last:
                chars[write_index] = char
                write_index += 1
                if count >= 2:
                    s = str(count)
                    for c in s:
                        chars[write_index] = c
                        write_index += 1
                count = 0
            index += 1

        return write_index
